shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -537,7 +558,7 @@ public void attachToFlutterEngine(@NonNull FlutterEngine flutterEngine) {
     textInputPlugin = new TextInputPlugin(
         this,
         this.flutterEngine.getDartExecutor(),
-        null
+        platformViewsController
```

Comment:
I see. Can we flip it around? Such that the FlutterView creates and owns the PlatformViewController and since the FlutterFragment only needs this in one place, it does 

```java
flutterEngine.getActivityControlSurface().attachToActivity(
          getActivity(),
          getLifecycle(),
          flutterView.getPlatformViewsController(),
)
```

Regarding the sequencing of onAttach->onCreate->onCreateView etc, I think we might want to tweak it a bit too. Otherwise it's a bit leaky that we're given an Activity which hasn't yet finished initializing to an activity aware plugin. It's a bit like giving a reference to a partially constructed object to someone else in Android equivalent.

If the flutterEngine.getActivityControlSurface().attachToActivity() was on onActivityCreated() instead, then it's both cleaner and onCreateView would have been called by then, meaning you can get the PlatformViewController from the FlutterView.

Reply:
I'll attempt to make this change and see how it goes.

Reply:
Actually, @xster, now I remember why the `PlatformViewsController` isn't created inside `FlutterView` - platform view functionality needs to outlive a given `FlutterView`, right? When the engine is detached from a view, we probably won't attempt to layout or render anything, but we probably can't just tear out the platform views subsystem just because we stopped rendering to the screen, can we?

Reply:
Doesn't this apply to the FlutterFragment (to which you're giving ownership of the platform view controller in this PR) as well?

Reply:
Also, I think moving flutterEngine.getActivityControlSurface().attachToActivity() from onAttach to onCreateView is onActivityCreated is a bit orthogonal. I just noticed it now but it wasn't really part of this PR. We should make it cleaner nevertheless either in this PR or in another one.

Reply:
@xster I just pushed a change that moves `PlatformViewsController` into `FlutterEngine` as a component. I think that conceptually makes sense since we've said that some notion of platform views needs to function even in the background. Let me know what you think.

WRT to `onActivityCreated`, we should chat about it. I looked up the Fragment lifecycle and I'm not convinced that there is a symmetric call. I think the question comes down to this: if I attach an existing Fragment to an existing Activity, will we get a call to `onActivityCreated`? If not, then I don't think we can use it.